lib/gollum-lib/filter/critic_markup.rb

```diff
@@ -0,0 +1,93 @@
+# ~*~ encoding: utf-8 ~*~
+
+# CriticMarkup
+#
+# Render CriticMarkup
+
+class Gollum::Filter::CriticMarkup < Gollum::Filter
+
+  # Patterns inspired by https://github.com/DivineDominion/criticmarkup.tmbundle/blob/master/Syntaxes/criticmarkup.tmLanguage
+  # All patterns use multiline matching (m flag)
+  # Logic inspired by https://github.com/CriticMarkup/CriticMarkup-toolkit/blob/master/CLI/criticParser_CLI.py
+  
+  ADDITION_PATTERN      = %r|{\+\+(?<content>.*?)\+\+[ \t]*(\[(.*?)\])?[ \t]*\}|m
+  DELETION_PATTERN      = %r|{--(?<content>.*?)--[ \t]*(\[(.*?)\])?[ \t]*\}|m
+  SUBSTITUTION_PATTERN  = %r|{~~(?<oldcontent>.*?)~>(?<newcontent>.*?)~~}|m
+  HIGHLIGHT_PATTERN     = %r|{\=\=(?<content>.*?)[ \t]*(\[(.*?)\])?[ \t]*\=\=\}{>>(?<comment>.*?)<<}|m
+  COMMENT_PATTERN       = %r|{>>(?<content>.*?)<<}|m
+
+  PROCESS_PATTERN       = /(?<placeholder>CRITIC\h+)/
+
+
+  def extract(data)
+    data.gsub! ADDITION_PATTERN do
+      content = $~[:content]
+      placeholder = "CRITIC" + Digest::SHA1.hexdigest("#{content}#{@map.size+1}")
```

Comment:
What's the rationale for adding `#{@map.size+1}` here?

Reply:
To make sure that the digest is unique.

Reply:
I suppose it could just be `@map.size`.

Reply:
Fixed in https://github.com/gollum/gollum-lib/pull/308/commits/23262da3fc1bac2be84c2c0ccebd1c4165474dc0.